pyccel/codegen/printing/fcode.py

```diff
@@ -694,21 +695,16 @@ def _print_NumpyWhere(self, expr):
     def _print_NumpyArray(self, expr):
         """Fortran print."""
 
-        # Always transpose indices because Numpy initial values are given with
-        # row-major ordering, while Fortran initial values are column-major
-        shape = expr.shape[::-1]
-
         # Construct right-hand-side code
-        if expr.rank > 1:
-            arg = functools.reduce(operator.concat, expr.arg)
-            rhs_code = 'reshape({array}, {shape})'.format(
-                    array=self._print(arg), shape=self._print(Tuple(*shape)))
-        else:
-            rhs_code = self._print(expr.arg)
-
+        rhs_code = self._print(expr.arg)
         # If Numpy array is stored with column-major ordering, transpose values
+        # use reshape with order for rank > 2
         if expr.order == 'F' and expr.rank > 1:
-            rhs_code = 'transpose({})'.format(rhs_code)
+            shape    = ', '.join(self._print(i) for i in expr.shape)
+            order    = [LiteralInteger(i) for i in range(1, expr.rank+1)]
+            order    = order[1:]+ order[:1]
+            order    = ', '.join(self._print(i) for i in order)
+            rhs_code = 'reshape({},[{}], order=[{}])'.format(rhs_code, shape, order)
 
         return rhs_code
```

Comment:
According to the comment `reshape` should be used for `rank > 2`, so I would expect `transpose` to be used when `rank == 2` instead.

Reply:
done

Reply:
Thanks